app/controllers/candidate_interface/prefill_application_form_controller.rb

```diff
@@ -54,5 +49,27 @@ def destroy_blank_application
       application_form = current_candidate.application_forms.first
       application_form.destroy if application_form.blank_application?
     end
+
+    def test_application_options
+      test_application_options = {
+        recruitment_cycle_year: RecruitmentCycle.current_year,
+        states: [:unsubmitted_with_completed_references],
+        courses_to_apply_to: Course.current_cycle.open_on_apply.joins(:course_options).merge(CourseOption.available),
+        candidate: current_candidate,
+      }
+
+      if prefill_application_state['prefill']
+        course_from_find = Course.find(prefill_application_state['course_id'])
+        test_application_options.merge!(courses_to_apply_to: [course_from_find])
```

Comment:
Should we clear the cache at this point?

Reply:
Yes agree